docs/integrations/databases/supabase.mdx

```diff
@@ -3,170 +3,666 @@ title: Integrate Supabase with Clerk
-/>
+- In your Supabase dashboard, in the sidebar, navigate to **Database** > **Tables**. From here, you can choose which table you want to edit, and add the new column.
+    - For this example, you can call the column `user_id`.
+    - **This column's data type must be `text`**.
 
 <Callout type="info">
-  Note: Supabase recommends enabling [Row Level Security (RLS)](https://supabase.com/docs/guides/auth/row-level-security) for your database tables and configuring access policies as needed.
+  This step is required becase Supabase's `auth.uuid()` function, which normally grants access to the user ID in RLS policies, is not compatible with Clerk's user IDs.
 </Callout>
 
-After setting those three environment variables, you should be able to start up your application development server.
-
-Install the JavaScript client for Supabase with:
-
-```sh filename="terminal"
-npm install @supabase/supabase-js
-```
-
-You can then initialize the Supabase client by passing it the environment variables and the access token from Clerk.
-
-```jsx
-import { useAuth } from '@clerk/clerk-react';
-import { createClient } from "@supabase/supabase-js";
-
-const supabaseClient = async (supabaseAccessToken) => {
-    const supabase = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL, process.env.NEXT_PUBLIC_SUPABASE_KEY, {
-      global: { headers: { Authorization: `Bearer ${supabaseAccessToken}` } },
-    });
-    // set Supabase JWT on the client object,
-    // so it is sent up with all Supabase requests
-    return supabase;
-  };
-  
-function App() {
-  const { getToken } = useAuth();
-
-  const fetchData = async () => {
-    // TODO #1: Replace with your JWT template name
-    const supabaseAccessToken = await getToken({ template: 'supabase' });
-
-    const supabase = await supabaseClient(supabaseAccessToken);
+### Create a SQL query that checks the user ID
+
+Next, you must create a `requesting_user_id()` function so you can access the user ID in your RLS policies:
+    1. In your Supabase dashboard, in the sidebar, navigate to **SQL Editor**, then select **New query**. Paste the following into the editor:
+        ```sql
+        create or replace function requesting_user_id() returns text as $$ select nullif(current_setting('request.jwt.claims', true)::json->>'sub', '')::text; $$ language sql stable;
+        ```
+    1. Select **Run** to execute the query and create the function.
+
+### Create ID-based RLS policies
+
+Now you must create RLS policies that allow users to modify and read content associated with their user IDs:
+1. Create an RLS policy for inserting content:
+    - In your Supabase dashboard, in the sidebar, navigate to **Authentication** > **Policies**. Under the name of the table you want users to have access to, select **New Policy**.
+    - If you're using the policy editor, paste the following snippet:
+        ```sql filename="Supabase policy editor"
+        CREATE POLICY "create user post" ON "public"."Posts"
```

Comment:
My thoughts here are we should probably use the original "Todos" example as it makes sense in context to scope todos to a specific user, whereas posts would only be scoped by user in a personal page for editing. 

I used Posts in my notes as a way to deviate from the original tutorial and see if it broke anything

Reply:
I checked out the blog post example, and to integrate that into this guide would require a pretty large rewrite. Instead, I changed it so this example fetches addresses scoped to a user. It's standard in ecommerce to have a separate addresses table when addresses are optional and users can have multiple